common/src/utils.rs

```diff
@@ -1,11 +1,26 @@
 use futures::Stream;
+use serde::{Deserialize, Serialize};
 use sp_core::ecdsa;
 use std::sync::atomic::AtomicBool;
 use std::sync::Arc;
 use tokio::sync::mpsc::UnboundedReceiver;
 
 pub const ECDSA_SIGNATURE_LENGTH: usize = 65;
 
+// constrain output types to have the `Deserialize` trait
+pub fn deserialize<'a, T>(data: &'a [u8]) -> Result<T, serde_json::Error>
+where
+    T: Deserialize<'a>,
+{
+    let msg = std::str::from_utf8(data).unwrap();
+    serde_json::from_str::<T>(msg)
```

Comment:
We should use `serde_json::from_slice` instead here. https://docs.rs/serde_json/latest/serde_json/fn.from_slice.html